### ISSUE TITLE
Task should not have setters

### ISSUE THREAD

**Issue by fhermeni** [chocoteam/choco-solver#383]
If you change a variable using a setter, then the monitor will not be plugged to the new variable, while the monitor over the old variable _might_ have to be removed.

For simplicity and robustness, why not removing the setters ?

**Comment by jgFages** [comment]
You are absolutely right. Actually, we have already removed them in the develop branch.

**Comment by fhermeni** [comment]
By the way, I had performance issues with tasks due to the constructor that make some tests to choose the appropriate monitor. I know that I am using bounded variables so by the end, I copy/pasted your code to just use the right monitor.

Maybe you should use the monitor as a constructor parameter and detect in VF. the one to choose. No changes for the lambda user but a better control for the advanced ones. You can also remove the attributes in Task and link the getters to getters in the monitor. This will remove redundant storage of attributes.

**Comment by jgFages** [comment]
Are you saying that an if statement with 3 conditions to check causes you performance issue?

**Comment by fhermeni** [comment]
> Le 3 avr. 2016 à 13:50, Jean-Guillaume Fages <notifications@github.com> a écrit :
> 
> Are you saying that an if statement with 3 conditions to check causes you performance issue?
> 


Yup
But as usual, this is a particular (but real) use case.
It involves tens of thousands of tasks and most of the variables are already instantiated. So the portion of this code is naturally magnified.

I think this static analysis should be done in VF (it looks like a factory pattern in fact). So I propose:
- Task as an interface
- Both monitors implement the interface
- VF choose the most suitable implementation
---
Fabien Hermenier
Associate Professor SCALE Project, University Nice Sophia Antipolis
http://fhermeni.github.io
